### PR TITLE
Activate dnsprovider replication controller

### LINKED ISSUE #57: Activate dnsprovider-replication controller
**How to categorize this issue?**

/area control-plane
/kind enhancement
/priority 3

**What would you like to be added**:
Activate the new dnsprovider-replication controller to allow to specify `DNSProvider` on shoot.
This is a follow-up for https://github.com/gardener/external-dns-management/issues/182

**Why is this needed**:

### PR DESCRIPTION
**Pull request by MartinWeindel**
**How to categorize this PR?**

/area control-plane
/kind enhancement

**What this PR does / why we need it**:
Allow to specify `DNSProvider` resources on the shoot and replicate them to the control plane. This is an alternative to specifying DNS providers in the shoot manifest and needs no access or permissions to the Garden project.

The replication is **disabled** by default, but can be enabled globally in the controller deployment by setting the value `dnsProviderReplication.enabled` to `true`.
It is also possible enable/disable the replication for a single shoot cluster in the providerConfig of the shoot-dns-service with the same parameter `dnsProviderReplication.enabled`.

**Which issue(s) this PR fixes**:
Fixes #57 

**Special notes for your reviewer**:

**Release note**:

```feature user
Support DNSProvider replication to allow to specify providers on shoot cluster.
```

### CODE REVIEW

File: charts/internal/shoot-dns-service-seed/templates/deployment.yaml [hunk 1]
```diff
@@ -48,14 +48,18 @@ spec:
...
+          - --dnsprovider-replication.target-realms={{ .Release.Namespace }},
+          {{- end }}
           - --namespace=kube-system
           - --target-namespace={{ .Release.Namespace }}
           - --target-creator-label-name=gardener.cloud/shoot-id
           - --target-creator-label-value={{ .Values.creatorLabelValue }}
           - --target-set-ignore-owners
           - --dns-target-class=gardendns
           - --dns-class={{ .Values.dnsClass }}
+          - --lease-name=shoot-dns-service
```
Comment (mandelsoft) [inline charts/internal/shoot-dns-service-seed/templates/deployment.yaml:62]: Does this require an rbac change?

Comment (MartinWeindel) [reply charts/internal/shoot-dns-service-seed/templates/deployment.yaml:62]: Currently the role covers all configmaps in the kube-system, so no change neeeded.
The dns-controller-manager still uses only configmaps for leases.

File: charts/internal/shoot-dns-service-shoot/values.yaml [hunk 2]
```diff
@@ -3,4 +3,7 @@
 # Declare variables to be passed into your templates.
 
 userName: dummy
-serviceName: dummy
\ No newline at end of file
+serviceName: dummy
+
+dnsProviderReplication:
+  enabled: true
```
Comment (timuthy) [inline charts/internal/shoot-dns-service-shoot/values.yaml:9]:
```suggestion
  enabled: false
```

**Review by timuthy — CHANGES_REQUESTED** [review]
Looks good in general, only one finding from my side.

**Review by timuthy — APPROVED** [review]
/lgtm

**Pull request merged**